### PR TITLE
[2.8] Deprecate IntrospectableContainerInterface

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | yes |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

Patch on Security is required for tests compat with 3.0, see #15684 
IntrospectableContainerInterface is already deprecated in master.

### CODE REVIEW

**Comment by Tobion** [comment]
ref https://github.com/symfony/symfony/issues/15346#issuecomment-124201286

**Comment by nicolas-grekas** [comment]
Thanks for the link.
Code wanting to be both 3.0 and 2.x compatible could rely on ContainerInterface + method_exists
Can we consider this?

**Comment by Tobion** [comment]
I agree

**Comment by nicolas-grekas** [comment]
I added 3.0 compat in Symfony\Bridge\ProxyManager\LazyProxy\PhpDumper\ProxyDumper

**Comment by Tobion** [comment]
:+1: 
Status: Reviewd

**Comment by Tobion** [comment]
Status: Reviewed

File: src/Symfony/Bundle/SecurityBundle/composer.json [hunk 1]
```diff
@@ -27,9 +27,7 @@
         "symfony/config": "~2.8|~3.0.0",
         "symfony/console": "~2.7|~3.0.0",
         "symfony/css-selector": "~2.0,>=2.0.5|~3.0.0",
-        "symfony/dependency-injection": "~2.6,>=2.6.6|~3.0.0",
         "symfony/dom-crawler": "~2.0,>=2.0.5|~3.0.0",
-        "symfony/form": "~2.8",
```
Comment (nicolas-grekas) [inline src/Symfony/Bundle/SecurityBundle/composer.json]: These two constraints are duplicates of similar ones in framework-bundle, and for some reason (a bug in composer?) they can create [dependency resolving failure](https://travis-ci.org/symfony/symfony/jobs/78934225#L2133). Removing them fixes the issue without changing anything to the resolved set.

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**